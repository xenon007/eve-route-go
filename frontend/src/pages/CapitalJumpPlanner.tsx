import React, { useState } from "react";
import { Box, Button, Grid, Typography } from "@material-ui/core";
import axios from "axios";
import SystemInput from "../components/SystemInput";

interface ApiResponse {
  route: Array<{ id: number; name: string }>;
}

/**
 * Страница планировщика прыжков капитальных кораблей.
 * Позволяет выбрать начальную и конечную системы
 * и рассчитать маршрут через API.
 */
export default function CapitalJumpPlanner() {
  const [start, setStart] = useState("");
  const [end, setEnd] = useState("");
  const [names, setNames] = useState<string[]>([]);
  const [message, setMessage] = useState("");

  const findRoute = () => {
    setMessage("");
    if (!start || !end) {
      return;
    }
    console.log("Requesting capital route", start, end);
    axios
      .get<ApiResponse>(`/api/capital?start=${start}&end=${end}`)
      .then((r) => {
        setNames(r.data.route.map((s) => s.name));
      })
      .catch((err) => {
        console.error("Failed to fetch capital route", err);
        setMessage("No route found");
      });
  };

  return (
    <Grid container spacing={2} className="card">
      <Grid item xs={12}>
        <Typography variant="h6" align="center">
          Capital Jump Planner
        </Typography>
      </Grid>

      <Grid item sm={5} xs={12}>
        <Box display="flex" justifyContent="center">
          <SystemInput
            fieldId="start-system"
            fieldName="Start"
            onChange={setStart}
            fieldValue={start}
            findRoute={findRoute}
          />
        </Box>
      </Grid>
      <Grid item sm={2} xs={12}></Grid>
      <Grid item sm={5} xs={12}>
        <Box display="flex" justifyContent="center">
          <SystemInput
            fieldId="end-system"
            fieldName="End"
            onChange={setEnd}
            fieldValue={end}
            findRoute={findRoute}
          />
        </Box>
      </Grid>

      <Grid item xs={12}>
        <Box display="flex" justifyContent="center">
          <Button
            variant="contained"
            color="primary"
            onClick={findRoute}
            disabled={!(start && end)}
          >
            Find Route
          </Button>
        </Box>
      </Grid>

      <Grid item xs={12}>
        {message && <Typography>{message}</Typography>}
        {!message && names.length > 0 && (
          <ol>
            {names.map((n, i) => (
              <li key={i}>{n}</li>
            ))}
          </ol>
        )}
      </Grid>
    </Grid>
  );
}
