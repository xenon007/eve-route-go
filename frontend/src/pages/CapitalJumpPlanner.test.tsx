import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import theme from "../theme";
import CapitalJumpPlanner from "./CapitalJumpPlanner";

test("CapitalJumpPlanner snapshot", () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <CapitalJumpPlanner />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
});
