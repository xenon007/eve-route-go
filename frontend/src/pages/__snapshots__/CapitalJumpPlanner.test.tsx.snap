// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capital snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root card MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
      >
        capital.title
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <input
          data-testid="start-system"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <input
          data-testid="end-system"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled=""
          id="find-route"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            capital.find
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
    >
      
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
    >
      <div
        data-testid="map"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
